src/models/MySpace.ts

```diff
@@ -36,32 +36,57 @@ export const MySpaceModel = {
     { userId, title, type }: AddWidgetInput,
     { db }: Context
   ): Promise<Widget> {
-    // For now, can only have one News widget
+    // Fetch all widgets so we can add new widgets in the correct order
+    const allWidgets = await this.get({ userId }, { db })
+    const guardianIdealWidget = allWidgets.find(
+      (s: Widget) => s.type === 'GuardianIdeal'
+    )
+    const featuredShortcutsWidget = allWidgets.find(
+      (s: Widget) => s.type === 'FeaturedShortcuts'
+    )
+
+    // For now, can only have one each of News, Guardian Ideal,
+    // and Featured Shortcuts widgets
     if (type === 'News') {
       const allWidgets = await this.get({ userId }, { db })
       const newsWidget = allWidgets.find((s: Widget) => s.type === 'News')
       if (newsWidget) throw new Error('You can only have one News section')
     }
 
     if (type === 'GuardianIdeal') {
-      const allWidgets = await this.get({ userId }, { db })
-      const guardianIdealWidget = allWidgets.find(
-        (s: Widget) => s.type === 'GuardianIdeal'
-      )
       if (guardianIdealWidget)
```

Comment:
Thought: maybe we should change this to `if (type === 'GuardianIdeal' && guardianIdealWidget)`?